### DISCUSSION TITLE
Change default sort order to time

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [jarun/nnn#1836] [Q&A]
Currently, I'm trying to change the default sort order to time, but I also want to keep my home directory sorted using version. I've set `NNN_ORDER` to v for `$HOME`, but it doesn't seem to work. I suspect it's getting overwritten due to the `nnn -Tt` command.

Steps to reproduce:

1. Set `NNN_ORDER` to v for `$HOME` in the nnn configuration.
2. Run `nnn -Tt` to set the sort order to time.

**Expected behavior:**
The home directory should be sorted using version, while other directories should be sorted by time.

**Actual behavior:**
All directories are sorted using time.

**Comment by N-R-K** [comment]
> I've set NNN_ORDER to v for $HOME

Show where and how you're setting it.

**Reply by limaceous-bushwhacker** [reply]
As an env var with `export NNN_ORDER="v:$HOME"`

**Reply by N-R-K** [reply]
> I suspect it's getting overwritten due to the nnn -Tt command.

This seems to be the case. I can reproduce as well.

**Comment by jarun** [comment]
The order priority is set here:

```
6882                         if ((!cfgsort[cfg.curctx] || (cfgsort[cfg.curctx] == 'c'))
6883                             && ((r = get_kv_key(order, path, maxorder, NNN_ORDER)) > 0)) { // NOLINT  
```

So when there is a clear user choice to order for a context, all directories follow that (including the ones having a order choice which is overridden).
Otherwise, we will never be able to change the order of a directory that has a preset order using `NNN_ORDER`.

Maybe we can have an option to prioritise `NNN_ORDER` always but somehow that feels like over-engineering to me.
